--- a/src/CurlDownloader.cpp
+++ b/src/CurlDownloader.cpp
@@ -67,7 +67,7 @@
 		return 0;
 	}
 
-	if(curl_easy_getinfo(dl->curl,CURLINFO_SPEED_DOWNLOAD,&speed)==CURLE_OK){
+	if(curl_easy_getinfo(dl->curl,CURLINFO_SPEED_DOWNLOAD_T,&speed)==CURLE_OK){
 		Json::Value info(Json::objectValue);
 		info["type"]="curl";
 		info["speed"]=speed;
@@ -223,7 +223,7 @@
 	curl_easy_setopt(curl, CURLOPT_WRITEDATA, this);
 	curl_easy_setopt(curl, CURLOPT_NOPROGRESS, 0);
 	curl_easy_setopt(curl, CURLOPT_PROGRESSDATA, this);
-	curl_easy_setopt(curl, CURLOPT_PROGRESSFUNCTION,CurlDownloader::CbProgress);
+	curl_easy_setopt(curl, CURLOPT_XFERINFOFUNCTION, CurlDownloader::CbProgress);
 
 	if(this->headersonly){
 		curl_easy_setopt(curl, CURLOPT_WRITEHEADER, this);

